### ISSUE TITLE
I don't know what to do with the unnamed block in the '' data entry

### ISSUE THREAD

**Issue by kepi** [bwanders/dokuwiki-strata#19]
I have clean install of strata plugin and I'm receiving following error every time I try to add more than items in data entry:

`I don't know what to do with the unnamed block in the '' data entry`

I'm trying with example from your documentation:

```
<data user>
Full Name: John Doe
Age: 24
Contact [link]: john.doe@example.org
Friends [ref]*: Alice, Bob
</data>
```
If I use only two items like:

```
<data user>
Full Name: John Doe
Age: 24
</data>
```
Than it works without any problem. I tried numerous variations of data.

When I disable line 87-90 in `syntax/entry.php` (sanity check) then everything seams ok.

It seams to me like there is some problem in `constructTree` method.

**Comment by kepi** [comment]
Ok, correction. It doesn't work OK with those lines commented. Third and fifth item is missing (at least from display).

I forgot to mention that I'm using PHP 7 (but no related errors in error log).

**Comment by bwanders** [comment]
Thanks for reporting this! Based on what you're experiencing this seems to something related to PHP 7.

Unfortunately, I'm currently swamped with work until the end of February. I'll pick this up at the start of March.

**Comment by nneul** [comment]
I'm also seeing this on our install since upgrading server to Ubuntu 16. Any update?

**Comment by nneul** [comment]
Additional bit - in my case it appears to be coming from the block in getFields() and not the sanity check.

**Comment by bwanders** [comment]
Can you post your Dokuwiki and PHP versions?

**Comment by kepi** [comment]
For me it was PHP 7.0.6 and Dokuwiki Detrius.

**Comment by nneul** [comment]
ii  libapache2-mod-php7.0              7.0.4-7ubuntu2.1                         amd64        server-side, HTML-embedded scripting language (Apache 2 module)

commit f847ae11f67fa5dd7c83151a0e1dc79f2e270278
Merge: e4261f2 42cbd32
Author: Andreas Gohr <andi@splitbrain.org>
Date:   Thu Apr 28 10:29:59 2016 +0200

    Merge pull request #1542 from splitbrain/protectedusers
    
    Add a protected file to the authplain config cascade

**Comment by nneul** [comment]
No change with dokuwiki latest

commit 33e99aba20eef5b012776ebc4db0b87284e62ac4
Author: Guy Brand <gb@unistra.fr>
Date:   Sun Jul 3 18:05:17 2016 +0200

    Release preparation

**Comment by bwanders** [comment]
Thanks for the feedback!

I'll be looking into this issue this week.

**Comment by bwanders** [comment]
Fixed with 61ae6f03a0770f70f35f6e6e6b5ef360f854453c.

The issue was indeed due to PHP7. In specific, the semantics of `foreach` have changed in an [incompatible manner](http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.foreach.by-ref). This led to some monkey-business during the handling of the syntax.
